src/PlaneWaveBasis.jl

```diff
@@ -302,3 +318,52 @@ end
 function r_to_G(basis::PlaneWaveBasis, kpt::Kpoint, f_real::AbstractArray3) where {T}
     r_to_G!(similar(f_real, length(kpt.mapping)), basis, kpt, copy(f_real))
 end
+
+#
+# Functions to handle BZ symmetry
+#
+
+"""
+Converts an eigenfunction at point `kpoint_src` to one at
+`kpoint_dst`, the two being related by symmetry operation S.
+"""
+function transfer_ψ(basis::PlaneWaveBasis{T}, kpoint_src::Kpoint, kpoint_dst::Kpoint, ψ::AbstractVector, S, τ) where {T}
+    @assert S*kpoint_src.coordinate ≈ kpoint_dst.coordinate
+    invS = Mat3{Int}(inv(S))
+    # ψsym(G) = e^{-i G τ} ψ(S^-1 G)
+    ψsym = zero(ψ)
+    for (iG, G) in enumerate(G_vectors(kpoint_dst))
+        igired = index_G_vectors(basis, kpoint_src, invS * G)
+        if igired !== nothing
+            @assert G_vectors(kpoint_src)[igired] ≈ invS*G
+            ψsym[iG] = cis(-2T(π)*dot(G, τ)) * ψ[igired]
+        end
+    end
+    ψsym
+end
+function transfer_ψ(basis::PlaneWaveBasis, kpoint_src::Kpoint, kpoint_dst::Kpoint, ψ::Tψ, S, τ) where {Tψ <: AbstractMatrix}
+    ψsym = zero(ψ)
+    for i = 1:size(ψ, 2)
+        @views ψsym[:, i] .= transfer_ψ(basis, kpoint_src, kpoint_dst, ψ[:, i], S, τ)
+    end
+    ψsym
+end
+
+# Convert a basis into one that uses or doesn't use BZ symmetrization
+# Mainly useful for debug purposes and in cases we don't want to
+# bother with symmetry
+function PlaneWaveBasis(basis::PlaneWaveBasis; enable_bzmesh_symmetry)
```

Comment:
I'm not sure I would make this a constructor. For me it feels more natural to just define a function `disable_kpoint_symmetry` or something and do it in there. Also gets rid of this strange kwarg.

Reply:
I actually quite like it like a constructor, it's what it is, no? And the kwarg is the same as the other constructors. The less extra functions we have, the less namespace pollution, and the less I have to remember stuff.

Reply:
Whether you not remember a name or not remember which of the 15 interfaces it was, what's the difference? It's pretty specific and low-level. It should not be honored to live on the same level as the high-level constructor with the MP-grid in my opinion.

Reply:
From my point of view the more constructors the better. There are some high level, some not. What distinguishes them is whether we use them in examples and whether we add a docstring to them (ie I wouldn't for this one). Having this kind of constructors is very convenient. Eg what if you want to make a pwbasis that has the same parameters as an existing one but with a larger fft_size? What if you want to do it as the same time as removing the bz symmetry? With this kind of constructors you can do it

Reply:
Ok, I see your point, but then we might need to clean it up a little from time to time. I have the feeling for PlaneWaveBasis it's getting out of hand.

Reply:
I agree pwbasis is a bit of a mess yeah...